client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-card/index.tsx

```diff
@@ -0,0 +1,93 @@
+import { Card, Gridicon } from '@automattic/components';
+import classNames from 'classnames';
+import { useState, ReactElement } from 'react';
+import SiteErrorContent from '../site-error-content';
+import SiteStatusContent from '../site-status-content';
+import type { ReactChild } from 'react';
+
+import './style.scss';
+
+interface Props {
+	rows: {
+		site: { value: { blog_id: number; url: string }; error: string; type: string; status: string };
+		scan: { threats: number };
+		plugin: { updates: number };
+		[ key: string ]: any;
+	};
+	columns: {
+		site: ReactChild;
+		backup: ReactChild;
+		scan: ReactChild;
+		monitor: ReactChild;
+		plugin: ReactChild;
+		[ key: string ]: any;
+	};
+}
+
+export default function SiteCard( { rows, columns }: Props ): ReactElement {
+	const [ isExpanded, setIsExpanded ] = useState( false );
+
+	const toggleIsExpanded = () => {
+		setIsExpanded( ! isExpanded );
+	};
+
+	const toggleContent = isExpanded ? (
+		<Gridicon icon="chevron-up" className="site-card__vertical-align-middle" size={ 18 } />
+	) : (
+		<Gridicon icon="chevron-down" className="site-card__vertical-align-middle" size={ 18 } />
+	);
+
+	const rowKeys = Object.keys( rows );
```

Comment:
I am little bit confused here. `rows` is an object with keys like `site`, `scan`, `plugin` etc. and `Object.keys( rows )` would return an array like `[ 'site', 'scan', 'plugin' ]`, but do we expected the array to be in certain order? `headerItem` below should be `'site'`. Is the sorting of the array guaranteed here?

Some unit tests may confirm our assumption.

Reply:
Very good catch, @manzoorwanijk. I have made the changes accordingly.